From 1253428c683a05cae4338baad89d4bef61ae6bb0 Mon Sep 17 00:00:00 2001
From: Amitesh Anand <quic_amitesh@quicinc.com>
Date: Wed, 27 Dec 2023 15:52:17 +0530
Subject: [PATCH 425/500] bonding: Set ieee80211_ptr only when kernel config is
 enabled.

Usage of net_device->ieee80211_ptr only when CONFIG_CFG80211
is enabled.

Change-Id: I264cd50e7cd65bf0e0665f1d3dcfb3f08dda295a
Signed-off-by: Amitesh Anand <quic_amitesh@quicinc.com>
---
 drivers/net/bonding/bond_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/bonding/bond_main.c b/drivers/net/bonding/bond_main.c
index 48c72317e11f..8de5de161b02 100644
--- a/drivers/net/bonding/bond_main.c
+++ b/drivers/net/bonding/bond_main.c
@@ -6867,7 +6867,9 @@ struct net_device *bond_create_mlo(struct net *net, const char *name, struct mlo
 	/*
 	 * MLO specific initialization.
 	 */
+#if IS_ENABLED(CONFIG_CFG80211)
 	bond_dev->ieee80211_ptr = mlo_info->wdev;
+#endif
 	bond->params.mode = BOND_MODE_MLO;
 	mlo_info->wdev->netdev = bond_dev;
 
-- 
2.34.1

